### ISSUE TITLE
Error on compile via "./fbt fap_bc_demo"

### ISSUE THREAD

**Issue by WILL-VESTERGAARD** [csBlueChip/FlipperZero_plugin_howto#4]
Upon attempting to compile the demo; with ./fbt fap_bc_demo.
The terminal returns:

```
PS C:\Users\MY_USER\flipperZero\official> ./fbt fap_bc_demo
        CC      applications_user\bc_demo\bc_demo.c
        SDKCHK  firmware\targets\f7\api_symbols.csv
applications_user\bc_demo\bc_demo.c: In function 'cbDraw':
applications_user\bc_demo\bc_demo.c:123:27: error: implicit declaration of function 'acquire_mutex' [-Werror=implicit-function-declaration]
  123 |  if ( !(state = (state_t*)acquire_mutex((ValueMutex*)ctx, 25)) )  return ;
      |                           ^~~~~~~~~~~~~
applications_user\bc_demo\bc_demo.c:123:42: error: 'ValueMutex' undeclared (first use in this function)
  123 |  if ( !(state = (state_t*)acquire_mutex((ValueMutex*)ctx, 25)) )  return ;
      |                                          ^~~~~~~~~~
applications_user\bc_demo\bc_demo.c:123:42: note: each undeclared identifier is reported only once for each function it appears in
applications_user\bc_demo\bc_demo.c:123:53: error: expected expression before ')' token
  123 |  if ( !(state = (state_t*)acquire_mutex((ValueMutex*)ctx, 25)) )  return ;
      |                                                     ^
applications_user\bc_demo\bc_demo.c:151:2: error: implicit declaration of function 'release_mutex' [-Werror=implicit-function-declaration]
  151 |  release_mutex((ValueMutex*)ctx, state);
      |  ^~~~~~~~~~~~~
applications_user\bc_demo\bc_demo.c:151:28: error: expected expression before ')' token
  151 |  release_mutex((ValueMutex*)ctx, state);
      |                            ^
applications_user\bc_demo\bc_demo.c: In function 'bc_demo':
applications_user\bc_demo\bc_demo.c:530:2: error: unknown type name 'ValueMutex'
  530 |  ValueMutex         mutex   = {0};
      |  ^~~~~~~~~~
applications_user\bc_demo\bc_demo.c:579:7: error: implicit declaration of function 'init_mutex' [-Werror=implicit-function-declaration]
  579 |  if (!init_mutex(&mutex, state, sizeof(state))) {
      |       ^~~~~~~~~~
applications_user\bc_demo\bc_demo.c:639:29: error: implicit declaration of function 'acquire_mutex_block' [-Werror=implicit-function-declaration]
  639 |    if ( !(state = (state_t*)acquire_mutex_block(&mutex)) ) {
      |                             ^~~~~~~~~~~~~~~~~~~
applications_user\bc_demo\bc_demo.c:710:11: error: request for member 'mutex' in something not a structure or union
  710 |  if (mutex.mutex) {
      |           ^
applications_user\bc_demo\bc_demo.c:711:3: error: implicit declaration of function 'delete_mutex' [-Werror=implicit-function-declaration]
  711 |   delete_mutex(&mutex);
      |   ^~~~~~~~~~~~
applications_user\bc_demo\bc_demo.c:712:8: error: request for member 'mutex' in something not a structure or union
  712 |   mutex.mutex = NULL;
      |        ^
cc1.exe: all warnings being treated as errors
scons: *** [build\f7-firmware-D\.extapps\bc_demo\bc_demo.o] Error 1
API version 19.0 is up to date

********** FBT ERRORS **********
build\f7-firmware-D\.extapps\bc_demo\bc_demo.o: Error 1
```

I am a total noob...

**Comment by csBlueChip** [comment]
Hi @RealDuckSauce 

Thanks for raising an issue on this :) ...It is not anything you have done ...Flipper apparently made "breaking changes" to the "core API" about 2 weeks ago.

The clue is `implicit declaration of function 'acquire_mutex'` ...and generally all the messages say something about "mutexes don't exist".

**I have converted the code to use the new mutex system. It should now work as expected**

Sadly. I have also noticed that any attempt to "beep" the speaker causes the FZ to reboot, so I have had to disable that feature (for now).

Hope this gets you up and running,
BC
